mighty/mighty_agents/base_agent.py

```diff
@@ -603,8 +606,23 @@ def run(  # noqa: PLR0915
                 metrics["episode_reward"] = episode_reward
 
                 action, log_prob = self.step(curr_s, metrics)
-                next_s, reward, terminated, truncated, _ = self.env.step(action)  # type: ignore
-                dones = np.logical_or(terminated, truncated)
+                # 1) step the env as usual
+                next_s, reward, terminated, truncated, infos = self.env.step(action)
+
+                # 2) decide which samples are true “done”
+                replay_dones = terminated          # physics‐failure only
+                dones    = np.logical_or(terminated, truncated)
+                
+
+                # 3) optionally overwrite next_s on truncation
+                if self.handle_timeout_termination:
```

Comment:
Not sure I like the naming. What does it mean to "handle_timeout_termination"? Should be more expressive. Also: when do we want this? Always? On specific envs? Specific algos? I would actually assume always, since we only want the next_s for next action prediction and always final obs in the replay. In that case we don't need a flag at all.